src/test/ERC721ShipyardRedeemablePreapprovedTraitSetters.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {ERC721ShipyardRedeemableOwnerMintable} from "./ERC721ShipyardRedeemableOwnerMintable.sol";
+import {ERC7498NFTRedeemables} from "../lib/ERC7498NFTRedeemables.sol";
+import {DynamicTraits} from "shipyard-core/src/dynamic-traits/DynamicTraits.sol";
+import {CampaignParams} from "../lib/RedeemablesStructs.sol";
+
+contract ERC721ShipyardRedeemablePreapprovedTraitSetters is ERC721ShipyardRedeemableOwnerMintable {
+    address[] public allowedTraitSetters;
```

Comment:
let's make this private, add a getter for getAllowedTraitSetters(), and make an onlyOwner setAllowedTraitSetters()